src/logging.cc

```diff
@@ -1110,7 +1110,7 @@ void LogFileObject::Write(bool force_flush,
        (bytes_since_flush_ >= 1000000) ||
        (CycleClock_Now() >= next_flush_time_) ) {
     FlushUnlocked();
-#ifdef OS_LINUX
+#if defined(OS_LINUX) && !defined(ANDROID)
```

Comment:
I do not know. I am compiling for android 4.x which does not have posix_fadvice. And it's a little soon to only support Android 5.x, as there are less than 25% Android 5.x users.